File: scaffolds/functions/ruby/handler.rb
# {name} — Flux function (compiled to WASM via ruby.wasm)
# Build: ruby.wasm build handler.rb -o {name}.wasm
require 'json'

# @param input_json [String]  JSON-encoded input payload
# @return [String]            JSON-encoded output
def {name}_handler(input_json)
  _input = JSON.parse(input_json)

  # TODO: implement {name}

  JSON.generate(ok: true)
end

File: scaffolds/functions/python/handler.py
# {name} — Flux function (compiled to WASM via py2wasm)
# Build: py2wasm -i handler.py -o {name}.wasm

import json

def handler(input_json: str) -> str:
    '''Entry point called by the Flux runtime.'''
    payload = json.loads(input_json)

    # TODO: implement {name}
    result = {"ok": True}

    return json.dumps(result)
